hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -37,6 +37,7 @@
 from hypothesis.strategies._internal.numbers import Real
 from hypothesis.strategies._internal.strategies import T, check_strategy
 from hypothesis.strategies._internal.utils import defines_strategy
+from numpy.typing import DTypeLike, NDArray
```

Comment:
I think this might increase our minimum supported numpy version.  That's fine if we _have to_, but I think we could probably use an `if TYPE_CHECKING:` block and string annotations to preserve compatibility?

Reply:
Right, I only checked for py37 compatibility.

NDArray was introduced in numpy 1.21 and DTypeLike in v1.20, v1.21 is from [june 2022](https://pypi.org/project/numpy/1.21.0/), I belive that's old enough,
I would personally introduce a `install_requires=['numpy>=1.21']` in the setup.py, but the call is totally yours.

Reply:
That's as old as [`NEP-29` suggests supporting](https://github.com/HypothesisWorks/hypothesis/blob/a147989cf5c936560528320c7a6a29a81daa5f9d/hypothesis-python/setup.py#L61), which seems a quite principled way to make the decision.  @honno do you have thoughts on this?

Regardless we should [set our actual minimum version in `extras_require`](https://github.com/HypothesisWorks/hypothesis/blob/a147989cf5c936560528320c7a6a29a81daa5f9d/hypothesis-python/setup.py#L61) - I doubt we actually still support back to 1.9.0 - and think about setting up a minimum-numpy-version CI job.  The CI job should be a separate PR though!

Reply:
> That's as old as [NEP-29 suggests supporting](https://github.com/HypothesisWorks/hypothesis/blob/a147989cf5c936560528320c7a6a29a81daa5f9d/hypothesis-python/setup.py#L61), which seems a quite principled way to make the decision. @honno do you have thoughts on this?

I missed that, then we are good to go, no need for `if TYPE_CHECKING` right?